src/main/java/com/eternalcode/parcellockers/database/JdbcConnectionProvider.java

```diff
@@ -0,0 +1,53 @@
+package com.eternalcode.parcellockers.database;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class JdbcConnectionProvider {
+
+    private final String dbUrl;
+    private final String user;
+    private final String pass;
+    private final String finalUrl;
+
+    public JdbcConnectionProvider(String dbUrl, String port, String databaseName, boolean useSSL, String user, String pass) {
+        this.dbUrl = dbUrl;
+        this.user = user;
+        this.pass = pass;
+        this.finalUrl = "jdbc:mysql://" + dbUrl + ":" + port + "/" + databaseName + "?useSSL=" + useSSL;
+    }
+
+    public Connection createConnection() {
+        try {
+            Class.forName("com.mysql.cj.jdbc.Driver");
+            return DriverManager.getConnection(this.finalUrl, this.user, this.pass);
+        }
+        catch (SQLException | ClassNotFoundException exception) {
+            throw new RuntimeException(exception);
+        }
+    }
+
+    public boolean executeUpdate(String sql) {
+        try (Connection connection = this.createConnection();
+            PreparedStatement statement = connection.prepareStatement(sql)) {
+            return statement.execute();
+        }
+        catch (SQLException exception) {
+            throw new RuntimeException(exception);
+        }
+    }
+
+    public ResultSet executeQuery(String sql) {
+        try (Connection connection = this.createConnection();
+             PreparedStatement statement = connection.prepareStatement(sql)) {
+            return statement.executeQuery();
```

Comment:
To działa? ResultSet nie jest usuwany razem z statement'em, który zostanie zamknięty razem z połączeniem?

Reply:
Nie wiem, nie testowałem...